File: python_programmers_school/level 1 review/짝수와 홀수.js
// num이 음수인 경우도 있음에 주의하자!
// 조건 연산자(삼항 연산자)를 사용한 풀이
function solution(num) {
  return Math.abs(num % 2) === 1 ? "Odd" : "Even";
}

// 좀 더 짧은 풀이
// num % 2의 결과가 0이 아닐 경우(참일 경우) Odd, 결과가 1일 경우(거짓일 경우) Even
// * 음수의 경우 아래와 같이 연산을 하면 나머지가 -0 or -1이 나온다. -0은 0과 완전히 같다.(===)
function evenOrOdd(num) {
  return num % 2 ? "Odd" : "Even";
}

File: python_programmers_school/level 1 review/짝수와 홀수.py
def solution(num):
    if num % 2:     # 2로 나눈 나머지가 0이 아닌 경우(True) (1인 경우)
        return "Odd"     # 홀수, return 되었으므로 함수 끝남.
    return "Even"     # 나머지가 0인 경우(False) 짝수

# js에서는 음수의 나머지는 음수로 나오는데, python에서는 음수의 나머지도 양수로 나온다.
# [js] 
# console.log(-5 % 2)     --> -1
# console.log(-4 % 2)     --> -0
# [python]
# print(-5 % 2)     --> 1
# print(-4 % 2)     --> 0